server/mailet/mock-smtp-server/src/test/java/org/apache/james/mock/smtp/server/MockSMTPBehaviorTest.java

```diff
@@ -94,4 +125,41 @@ void constructorShouldThrowWhenNumberOfAnswersIsNull() {
         assertThatThrownBy(() -> new MockSMTPBehavior(SMTPCommand.NOOP, Optional.empty(), RESPONSE, null))
             .isInstanceOf(NullPointerException.class);
     }
+
+    @Nested
+    class JSONTest {
+        @Test
+        void jacksonShouldDeserializeMockSMTPBehaviorWithAllField() throws Exception {
+            MockSMTPBehavior behavior = OBJECT_MAPPER.readValue(JSON_ALL_FIELDS, MockSMTPBehavior.class);
+
+            assertThat(behavior)
+                .isEqualTo(POJO_ALL_FIELDS);
+        }
+
+        @Test
+        void jacksonShouldSerializeMockSMTPBehaviorWithAllField() throws Exception {
+            String json = OBJECT_MAPPER.writeValueAsString(POJO_ALL_FIELDS);
+
+            assertThatJson(json)
+                .withOptions(new Options(Option.TREATING_NULL_AS_ABSENT))
+                .isEqualTo(JSON_ALL_FIELDS);
+        }
+
+        @Test
+        void jacksonShouldDeserializeMockSMTPBehaviorWithCompulsoryField() throws Exception {
+            MockSMTPBehavior behavior = OBJECT_MAPPER.readValue(JSON_COMPULSORY_FIELDS, MockSMTPBehavior.class);
+
+            assertThat(behavior)
+                .isEqualTo(POJO_COMPULSORY_FIELDS);
+        }
+
+        @Test
+        void jacksonShouldSerializeMockSMTPBehaviorWithCompulsoryField() throws Exception {
+            String json = OBJECT_MAPPER.writeValueAsString(POJO_COMPULSORY_FIELDS);
+
+            assertThatJson(json)
+                .withOptions(new Options(Option.TREATING_NULL_AS_ABSENT))
```

Comment:
Not 100% sure but there is a way I think to set up that directly with the ObjectMapper setup?

Reply:
that's some assertThatJson setup not some Jackson setup